src/replication.c

```diff
@@ -1914,6 +1943,169 @@ disconnect_nonresponding_replica(replica_t *replica, uint64_t io_seq,
 	MTX_UNLOCK(&replica->r_mtx);
 }
 
+static void
+handle_snap_list_response(replica_t *replica, void *resp_data, mgmt_cmd_t *mgmt_cmd, uint64_t len)
+{
+	zvol_io_hdr_t *hdr = replica->mgmt_io_resp_hdr;
+	struct zvol_snapshot_list *snap_details;
+	rcommon_mgmt_cmd_t *rcomm_mgmt = mgmt_cmd->rcomm_mgmt;
+	bool delete = false, assigned, failed = false;
+	struct zvol_snapshot_list **snap_resp_array = (struct zvol_snapshot_list **)rcomm_mgmt->buf;
+	uint64_t i = 0;
+
+	snap_details = calloc(1, len);
+	memcpy(snap_details, resp_data, len);
```

Comment:
why this? why not just type cast resp_data and use it.

Reply:
this can be freed by replica_response_handler if caller_gone is set. This is with new changes.